packages/cli-repl/src/cli-repl.ts

```diff
@@ -383,7 +426,9 @@ class CliRepl {
    */
   // eslint-disable-next-line @typescript-eslint/require-await
   async getConfig<K extends keyof CliUserConfig>(key: K): Promise<CliUserConfig[K]> {
```

Comment:
Maybe a bit of a silly question, but should the CliRepl class implement the `ConfigProvider` interface?

Reply:
Not a silly question at all – It should implement `MongoshIOProvider`, which includes a number of properties from `ConfigProvider`. :slightly_smiling_face: Typescript should already error out if it doesn’t do so, but I also don’t see any reason not to make the `implements` relationship explicit – done in 96d760033adfad588fb60f7f0307688692b4649c :)

Reply:
Ah cool makes sense. Nice add to make it explicit.